TicTacToe-GUI/TicTacToe.py

```diff
@@ -0,0 +1,230 @@
+    #check for asc win
+    if board[2][0]==player and board[1][1]==player and board[0][2]==player:
+        draw_asc_diagonal(player)
+        return True
+    if board[0][0]==player and board[1][1]==player and board[2][2]==player:
+        draw_des_diagonal(player)
+        return True
+
+def draw_horizontal_winning_line(row,player):
+    posY=row*square_size +square_size//2
+    
+    if(player==1):
+        color=circle_color
+    else:
+        color=cross_color
+    
+    pygame.draw.line(screen, color, (15,posY), (width-15,posY),15)
+    
+    
+    
+def draw_vertical_winning_line(col,player):
+    posX=col*square_size +square_size//2
+    if(player==1):
+        color=circle_color
+    else:
+        color=cross_color
+    
+    pygame.draw.line(screen, color, (posX,15), (posX,width-15),15)
+def draw_asc_diagonal(player):
+    if(player==1):
+        color=circle_color
+    else:
+        color=cross_color
+        
+    pygame.draw.line(screen,color,(15,height-15),(width-15,15),15)
+def draw_des_diagonal(player):
+    if(player==1):
+        color=circle_color
+    else:
+        color=cross_color
+        
+    pygame.draw.line(screen,color,(15,15),(width-15,height-15),15)
+
+
+
+def restart():
+    screen.fill(bg_color)
+    draw_lines()
+    player = 1
+    
+    for row in range(board_rows):
+        for col in range(board_columns):
+            board[row][col]=0
+    
+       
+draw_lines()
+#player
+player=1
+game_over=False
+
+while True: # main game loop
+    for event in pygame.event.get(): #constantly looks for the event
+        if event.type == pygame.QUIT: #if user clicks exit pygame.QUIT and sys exits
+            pygame.quit()
+            sys.exit()
+            
+        board_full=is_board_full()
+        if board_full and not game_over:
+           Won= font.render("    It's a Tie    ", True ,blue,green)
+           screen.blit(Won, winRect)
+           screen.blit(text, textRect) 
+           screen.blit(leave,leaveRect)
+        
+        if event.type==pygame.MOUSEBUTTONDOWN and not game_over:
+            mouseX= event.pos[0] #x
+            mouseY= event.pos[1] #y
+            
+            clicked_row=int(mouseY // square_size)
+            
+            clicked_column=int(mouseX // square_size)
+            
+            if available_square(clicked_row, clicked_column):
+                mark_square(clicked_row,clicked_column, player)
+                if(check_win(player)):
+                    game_over=True
+                    Won= font.render("Player"+str(player)+" Won ", True ,blue,green)
+                    screen.blit(Won, winRect)
+                    screen.blit(text, textRect) 
+                    screen.blit(leave,leaveRect)
+                player=player%2 +1
+                if not game_over and not board_full:
+                    Won= font.render("Player"+str(player)+" Turn ", True ,blue,green)
+                    screen.blit(Won, winRect)
+                draw_figures()
+        

```

Comment:
Please remove the extra blank lines

Reply:
@Lakhankumawat Please make these changes so that we can move your PR to PA